packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -468,23 +487,34 @@ export class VpcSubnet extends VpcSubnetRef {
      */
     public readonly subnetId: VpcSubnetId;
 
+    /**
+     * Manage tags for Construct and propagate to children
+     */
+    public readonly tags: cdk.TagManager;
+
     /**
      * The routeTableId attached to this subnet.
      */
     private readonly routeTableId: cdk.Token;
 
     constructor(parent: cdk.Construct, name: string, props: VpcSubnetProps) {
         super(parent, name);
+        this.tags = new cdk.TagManager(this, props.tags);
+        if (!this.tags.hasTag('Name')) {
+            this.tags.addTags({Name: {value: this.path, propagate: false}});
```

Comment:
I wonder if this is what we really want here? Why won't the various sub-resources have a `Name` based on their parent?

Perhaps you want something like: `addTag(key, value, { propagate: true, overwrite: false })`

Reply:
I am trying to understand the `overwrite` aspect here. Are you saying that if I have a tag I want to default (e.g. Name) and I find a propagated Name with `overwrite: false` I would set the value to my default? If I find `overwrite: true` I would not set the value with my default? 

My first thought on this particular use case is that propagating `Name` for subnet actually does make sense and we should change to `propagate: true`. Instead of `overwrite` becoming a property, couldn't I just checked the passed in tags (`props.tags`) instead of trying to figure out propagated and overwrite? If Name isn't passed in add it, otherwise the user wanted a specific name.

Reply:
I agree that propagating `Name` for subnets make sense in this case, so let's just do that and think about `overwrite` later (the question is whether propagation is "forced" or "optional", but if we don't have a clear use case right not, let's punt).

Reply:
I ended up adding a method `hasLocalTag(key)` that I think allows us to punt on this.